src/components/UserSearch.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react';
+import {Route} from 'react-router-dom';
+
+class UserSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      githubUsername: '',
+      selectedUsername: '',
+    };
+
+    this.searchUser = this.searchUser.bind(this);
+    this.updateUsername = this.updateUsername.bind(this);
+  }
+
+  searchUser(evt) {
+    evt.preventDefault();
+    // TODO: Dispatch action that will request data from githubs API
+    console.log("Searching user....");
+    this.setState({selectedUsername: this.state.githubUsername});
+  }
+
+  updateUsername(evt) {
+    console.log(evt);
+    this.setState({
```

Comment:
Also, keep in mind: once you connect a component, you limit its re-usability. I say this because I caution you to **not** export this component as already connected. Searching for a user by their username can be re-used without nailing it to this specific use-case.